contracts/src/contracts/hooks/aggregation_hook.cairo

```diff
@@ -0,0 +1,92 @@
+#[starknet::contract]
+pub mod aggregation_hook {
+    use alexandria_bytes::{Bytes, BytesTrait};
+    use hyperlane_starknet::contracts::hooks::libs::standard_hook_metadata::standard_hook_metadata::{
+        StandardHookMetadata, VARIANT,
+    };
+    use hyperlane_starknet::contracts::libs::message::Message;
+    use hyperlane_starknet::interfaces::{
+        IPostDispatchHook, Types, IPostDispatchHookDispatcher, IPostDispatchHookDispatcherTrait
+    };
+    use starknet::{ContractAddress};
+
+    #[storage]
+    struct Storage {
+        hooks: LegacyMap::<usize, ContractAddress>,
+        hook_count: usize,
+    }
+
+    mod Errors {
+        pub const INVALID_METADATA_VARIANT: felt252 = 'Invalid metadata variant';
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, hooks: Span<ContractAddress>) {
```

Comment:
Hooks are not meant to be defined in the constructor but retrieved from the metadata. But this might be a little tricky, so I can handle it if necessary